src/dataset.cpp

```diff
@@ -61,16 +61,27 @@ VariableSlice Dataset::operator()(const Tag tag, const std::string &name) & {
 }
 
 void Dataset::insert(Variable variable) {
-  for (const auto &item : m_variables)
-    if (item.tag() == variable.tag() && item.name() == variable.name())
-      throw std::runtime_error(
-          "Attempt to insert variable with duplicate tag and name.");
   // TODO special handling for special variables types like
   // Data::Histogram (either prevent adding, or extract into underlying
   // variables).
-  mergeDimensions(variable.dimensions(),
-                  coordDimension[variable.tag().value()]);
-  m_variables.push_back(std::move(variable));
+  if (contains(variable.tag(), variable.name())) {
+    auto &old = m_variables[find(variable.tag(), variable.name())];
+    for (const auto dim : old.dimensions().labels()) {
+      bool found = false;
+      for (const auto &var : m_variables)
+        if (var.dimensions().contains(dim))
+          found = true;
+      if (!found)
+        m_dimensions.erase(dim);
```

Comment:
So this is going to remove the dimensions supported by the old variable, that are not used in any other variable? 
Also, doesn't this self compare with `old`? i.e. condition on line 72  will return true for one comparison?

Reply:
Right, fixed.